### PR TITLE
Add TS Linting to the Major projects

### PR DESCRIPTION
**Pull request by emadelwany**

### CODE REVIEW

File: packages/sdk/client/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "built/index.js",
   "types": "built/index.d.ts",
   "scripts": {
-    "build": "npm run clean && tsc",
+    "build": "npm run clean && tsc && tslint --project tsconfig.json --force",
```
Comment (compulim) [inline packages/sdk/client/package.json]: We want to run `npm run build -- --watch` (or similar scripts) to build continuously. I don't know what's the best way to make it work, it seems no one find it yet.

Some people use `npm run watch`, some people use `npm run build:watch`, some use `npm run build -- --watch`. 😕

File: packages/sdk/main/package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "built/index.js",
   "types": "built/index.d.ts",
   "scripts": {
-    "build": "npm run clean && tsc",
+    "build": "npm run clean && tsc && tslint --project tsconfig.json --force",
```
Comment (compulim) [inline packages/sdk/main/package.json]: Consider `--watch`.

File: packages/sdk/shared/package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "built/index.js",
   "types": "built/index.d.ts",
   "scripts": {
-    "build": "npm run clean && tsc",
+    "build": "npm run clean && tsc && tslint --project tsconfig.json --force",
```
Comment (compulim) [inline packages/sdk/shared/package.json]: Consider `--watch`.

File: packages/app/client/webpack.config.js [hunk 4]
```diff
@@ -32,7 +32,12 @@ const defaultConfig = {
         test: /\.(tsx?)|(jsx)$/,
         exclude: [/node_modules/],
         use: ['awesome-typescript-loader']
-      }
+      },
+      {
+        test: /\.tsx?$/,
+        loader: 'tslint-loader',
+        options: { /* Loader options go here */ }
+      },
```
Comment (compulim) [inline packages/app/client/webpack.config.js]: Trailing commas.

File: packages/app/main/gulpfile.js [hunk 5]
```diff
@@ -128,12 +129,25 @@ gulp.task('build-extensions',
     'build-debug-extension')
 );
 
+//----------------------------------------------------------------------------
+gulp.task("tslint-app", () =>
```
Comment (compulim) [inline packages/app/main/gulpfile.js]: 2-space indentation.

File: packages/app/client/tslint.json [hunk 6]
```diff
@@ -1 +1,97 @@
-{ "rules": { "no-var-keyword": false } }
+{
```
Comment (compulim) [inline packages/app/client/tslint.json]: 2-space indent?

File: packages/sdk/client/package.json (hunk 1 above)
Comment (justinwilaby) [reply packages/sdk/client/package.json]: We're using webpack-dev-server and HMR for regular development. Some IDEs (at least mine) has real-time tslinting services built in so there's no need to have a script that does this.,

File: packages/sdk/main/package.json (hunk 2 above)
Comment (justinwilaby) [reply packages/sdk/main/package.json]: I can setup HMR for the main process. I came across some example configs for this but skipped over them in favor of focusing on the client.

**Comment by justinwilaby** [comment]
Eventually, I'd like to:
1. remove Gulp for builds - node/webpack can very easily handle this without the dependency on Gulp
2. Host a single tsconfig ,tslint and webpack.config at the root that will build all projects
3. implement HMR for the main process.

File: packages/sdk/main/package.json (hunk 2 above)
Comment (emadelwany) [reply packages/sdk/main/package.json]: I opened #589 to enable HMR in main

**Pull request merged**